homeassistant/components/iotty/__init__.py

```diff
@@ -0,0 +1,60 @@
+"""The iotty integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+
+from iottycloud.device import Device
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.config_entry_oauth2_flow import (
+    OAuth2Session,
+    async_get_config_entry_implementation,
+)
+
+from . import coordinator
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+type IottyConfigEntry = ConfigEntry[KnownDevicesData]
+
+
+@dataclass
+class KnownDevicesData:
+    """Contains information useful for handling run-time changes in case of added or removed devices."""
+
+    known_devices: set[Device]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up iotty from a config entry."""
+    _LOGGER.debug("async_setup_entry entry_id=%s", entry.entry_id)
+
+    implementation = await async_get_config_entry_implementation(hass, entry)
+    session = OAuth2Session(hass, entry, implementation)
+
+    data_update_coordinator = coordinator.IottyDataUpdateCoordinator(
+        hass, entry, session
+    )
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = data_update_coordinator
+    entry.runtime_data = KnownDevicesData(set())
```

Comment:
Is there a reason to store a reference to the coordinator in `hass.data`, or can we store both in the config entry's runtime data?

Reply:
Hi, looking at other examples (and even in the documentation) this looks like is the standard, so that's the main reason why it's inside hass.data. Do you see any drawbacks in that?
I think it could also be in runtime_data, though it looks a bit less convenient to access the two that way